app/components/channel_drawer/channels_list/channel_item/index.js

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import {connect} from 'react-redux';
+
+import {getCurrentChannelId, getChannel} from 'mattermost-redux/selectors/entities/channels';
+import {getTheme} from 'mattermost-redux/selectors/entities/preferences';
+
+import ChannelItem from './channel_item';
+
+function mapStateToProps(state, ownProps) {
+    const channel = getChannel(state, ownProps.channelId);
+
+    return {
+        currentChannelId: getCurrentChannelId(state),
+        displayName: channel.display_name,
+        status: channel.status,
+        theme: getTheme(state),
+        type: channel.type
+    };
+}
+
+function areStatesEqual(next, prev) {
```

Comment:
This doesn't seem necessary here. It'll also break the UI if the user changes their theme or name display setting